spec/services/path_generator_service_spec.rb

```diff
@@ -0,0 +1,30 @@
+# frozen_string_literal: true
+
+require 'spec_helper'
+
+RSpec.describe PathGeneratorService do
+  describe '.path' do
+    it "generates a base path which is unique to our database" do
+      service = PathGeneratorService.new
+      original_document = create(:document)
+      new_document = build(:document, document_type: original_document.document_type)
+      publishing_api_has_lookups("#{original_document.base_path}": nil)
+      expect(service.path(new_document, original_document.title)).to eq("#{original_document.base_path}-1")
+    end
+
+    it "raises a 'Path unable to be generated' when many variations of that path are in use" do
+      service = PathGeneratorService.new(5)
```

Comment:
Could we make this '2' and make the test shorter? I don't feel like we're proving anything special with a bigger number.